File: Sem3/Advanced Methods of Programming/MyInterpreter - CSharp Console Application/MyInterpreter CSharp/domain/statement/CompoundStatement.cs
using MyInterpreter_CSharp.domain.adt;

namespace MyInterpreter_CSharp.domain.statement
{
    public class CompoundStatement : IStatement
    {
        private IStatement _first;
        private IStatement _second;

        public CompoundStatement(IStatement first, IStatement second)
        {
            _first = first;
            _second = second;
        }


        public ProgramState Execute(ProgramState programState)
        {
            IMyStack<IStatement> stack = programState.ExeStack;
            stack.Push(_second);
            stack.Push(_first);
            return null;
        }

        public override string ToString()
        {
            return "(" + _first + ";" + _second + ")";
        }
    }
}

File: Sem3/Advanced Methods of Programming/MyInterpreter - JavaFX Graphical Application/project/src/domain/statement/CompoundStatement.java
package domain.statement;


import domain.ProgramState;
import domain.adt.MyStackInterface;


public class CompoundStatement implements StatementInterface
{
    private StatementInterface first;
    private StatementInterface second;


    public CompoundStatement(StatementInterface first, StatementInterface second)
    {
        this.first = first;
        this.second = second;
    }


    @Override
    public String toString()
    {
        return "(" + first.toString() + ";" + second.toString() + ")";
    }


    @Override
    public ProgramState execute(ProgramState programState)
    {
        MyStackInterface<StatementInterface> stack = programState.getExeStack();
        stack.push(second);
        stack.push(first);
        return null;
    }
}
